tests/validation_v2/upgrade/rancher_upgrade_test.py

```diff
@@ -0,0 +1,117 @@
+# How to Launch:
+# Example: <validation-tests>/v2/tests/upgrade/rancher_upgrade_test.py -b 0.37.0 -t 0.37.1 -s 104.197.121.156 -u aruneli
+# base version has no significance now, but target version should be a valid rancher server version
+# server option (-s): IP Address of the rancher server
+# username is the username with which you ssh to your GCE instance
+
+from tests.validation_v2.cattlevalidationtest.core.common_fixtures import *  # NOQA
+import argparse
+import os
+import paramiko
+import requests
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-b', help='base version')
+    parser.add_argument('-t', help='target version')
+    parser.add_argument('-s', help='server node')
+    parser.add_argument('-u', help='ssh username of rancher server host')
+    args = parser.parse_args()
+
+    tmp_dir = os.path.join(root_dir, 'tmp')
+    logger.info("tmp_dir is: %s", tmp_dir)
+
+    core_dir = os.path.join(root_dir,'tests','validation','cattlevalidationtest','core')
+    logger.info("core_dir is: %s", core_dir)
+
+    core_target_checkedout_dir = os.path.join(tmp_dir,'rancher-tests','tests','validation','cattlevalidationtests','core')
+    logger.info("core_target_checkedout_dir is: %s", core_target_checkedout_dir)
+
+    core_target_dir = os.path.join(root_dir,'tests','validation','cattlevalidationtest','core_target')
+    logger.info("core_target_dir is: %s", core_target_dir)
+
+    upgrade_test(args.b, args.t, args.s, args.u, tmp_dir, core_dir, core_target_dir, core_target_checkedout_dir)
+
+
+def upgrade_test(base, target, servernode, username, tmp_dir, core_dir, core_target_dir, core_target_checkedout_dir):
+    logger.info("CREATING SERVICES NOW IN BASE SETUP...")
+    create_cmd = "py.test "+core_dir+"/ -v -m create -s"
+    logger.info("create command is: %s", create_cmd)
+
+    os.system(create_cmd)
+    time.sleep(5)
+
+    upgrade_rancher_server(base, target, servernode, username)
+    #below one line is temporary until we start having real tagged versions
+    os.system(("cp -r "+core_dir+"/*.py "+tmp_dir))
+    #os.system("git clone -b master --single-branch https://github.com/aruneli/rancher-tests.git")
+    logger.info("COPYING TARGET LIBRARIES in core_target folder...")
+
+    os.system(("cp -r "+tmp_dir+"/*.py "+core_target_dir))
+    logger.info("VALIDATING UPGRADED SETUP NOW WITH TARGET")
+
+    validate_cmd = "py.test "+core_target_dir+"/ -v -m validate -s"
+    logger.info(validate_cmd)
+
+    os.system(validate_cmd)
+    logger.info("VALIDATION COMPLETE")
+
+    os.system("rm -rf "+tmp_dir+"/*")
+    os.system("rm -rf "+core_target_dir+"/*")
+
+
+def upgrade_rancher_server(base, target, servernode, username):
```

Comment:
rancher_upgrade_test.py is simple util script outside of validation tests purpose. Optimizations like moving the common code to utils is not done yet. This PR is mainly to merge the v2 framework so that others can start adding new test cases in new format. I am not done with code optimizations which is not of high priority now. Please review only rancher_upgrade_test.py, common_fixtures.py and the format of new test cases. Once a complete run is done in a GCE host as you requested, you can merge.